src/node.rs

```diff
@@ -93,6 +94,31 @@ impl<'a, T: Allocator> Node<'a, T> {
     }
 }
 
+impl<'a, T: Allocator> PartialEq for Node<'a, T> {
+    fn eq(&self, other: &Self) -> bool {
+        match (self.sexp(), other.sexp()) {
+            (SExp::Pair(l0, l1), SExp::Pair(r0, r1)) => {
```

Comment:
This recursion may bite us for deep trees.

Reply:
agree. I'm not sure why it didn't actually, probably because the stack frame is smaller